### PR TITLE
improved in out degree performance

### PR DESCRIPTION
**Pull request by swilly22**
#2753
`MATCH (a) RETURN outDegree(a), inDegree(a)` 

Previous implementation would extract `a's` incoming / outgoing edges into an array just to get its length and soon after discard it which is a waste. This new implementation doesn't extracts any edges but simply counts.

FYI with @AviAvni new design of NxM Relation matrices (N #nodes, M #edges) we get rid of Multi edge arrays and can simplify this change to a straight forward row reduction.

Quick benchmark:
Compute out degree of a node with 1M outgoing edges

```cypher
GRAPH.QUERY g "CREATE (a:A), (b:B)"
GRAPH.QUERY g "MATCH (a:A), (b:B) WITH a, b UNWIND range (0, 1000000) as x CREATE (a)-[:E]->(b)"
GRAPH.QUERY g "MATCH (a:A) RETURN outdegree(a)"
```

This PR 0.2ms
Current `Master` 20ms
100X improvement

### CODE REVIEW

**Review by vityafx — APPROVED** [review]
As far as I understand, looks good enough to me, and quite a simple change in the code.

File: src/arithmetic/entity_funcs/entity_funcs.c [hunk 1]
```diff
@@ -140,43 +140,67 @@ SIValue AR_EXISTS(SIValue *argv, int argc, void *private_data) {
...
 			Schema *s = GraphContext_GetSchema(gc, label, SCHEMA_EDGE);
-			if(!s) continue;
+			if(s == NULL) {
+				continue;
+			}
 
-			// Accumulate edges.
-			Graph_GetNodeEdges(gc->g, n, dir, s->id, &edges);
+			// count edges
+			count += Graph_GetNodeDegree(gc->g, n, dir, s->id);
```
Comment (vityafx) [inline src/arithmetic/entity_funcs/entity_funcs.c:168]: I can't find the definition for this function for some reason. Did you forget to add/commit/push it? :-)

Comment (raz-mon) [reply src/arithmetic/entity_funcs/entity_funcs.c:168]: @vityafx See graph.c

Comment (vityafx) [reply src/arithmetic/entity_funcs/entity_funcs.c:168]: I don't have it, perhaps, I am on an old master.

Comment (vityafx) [reply src/arithmetic/entity_funcs/entity_funcs.c:168]: Oh, weird! It wasn't showing up here for some reason. I checked a few times! I can see it here now in this PR.

File: src/graph/graph.c [hunk 2]
```diff
@@ -902,6 +902,114 @@ void Graph_GetNodeEdges
...
+(
+	const Graph *g,      // graph to inquery
+	const Node *n,       // node to get degree of
+	GRAPH_EDGE_DIR dir,  // incoming/outgoing/both
+	int edgeType         // relation type
+) {
+	ASSERT(g != NULL);
+	ASSERT(n != NULL);
+
+	GrB_Info info;
```
Comment (vityafx) [inline src/graph/graph.c:916]: I can't understand the point of having the "info" variable defined if it is unused.

File: src/graph/graph.c [hunk 3]
```diff
@@ -902,6 +902,114 @@ void Graph_GetNodeEdges
...
+
+	// for each relationship type to consider
+	for(edgeType = start_rel; edgeType < end_rel; edgeType++) {
+		M = Graph_GetRelationMatrix(g, edgeType, false);
+
+		//----------------------------------------------------------------------
+		// outgoing edges
+		//----------------------------------------------------------------------
+
+		// TODO: revisit once we get rid of MULTI-EDGE hack
```
Comment (vityafx) [inline src/graph/graph.c:959]: Should we anticipate any changes regarding this "TODO" in this PR or later?

**Review by vityafx — APPROVED** [review]

**Review by raz-mon — APPROVED** [review]

File: src/graph/graph.c (hunk 3 above)
Comment (swilly22) [reply src/graph/graph.c:956]: probably later down the road

**Pull request merged**